Sources/Profile-View.php

```diff
@@ -1885,33 +1885,38 @@ function TrackIP($memID = 0)
 	}
 	else
 	{
-		$context['ip'] = $user_profile[$memID]['member_ip'];
+		$context['ip'] = ip2range($user_profile[$memID]['member_ip']);
 		$context['base_url'] = $scripturl . '?action=profile;area=tracking;sa=ip;u=' . $memID;
 	}
 
 	// Searching?
 	if (isset($_REQUEST['searchip']))
-		$context['ip'] = trim($_REQUEST['searchip']);
+		$context['ip'] = ip2range(trim($_REQUEST['searchip']));
 
-	if (isValidIP($context['ip']) === false)
+	if (count($context['ip']) != 2)
```

Comment:
!==

Reply:
Can you explain?

Reply:
See below, but basically your comparing an integer to an integer, therefore you can use strict type comparision (!==/===) (good practice).

Reply:
But which issue solve this?